tests/dragonfly/config_test.py

```diff
@@ -0,0 +1,26 @@
+import pytest
+import redis
+from redis.asyncio import Redis as RedisClient
+from .utility import *
+from . import DflyStartException
+
+
+async def test_maxclients(df_factory):
+    # Needs some authentication
+    server = df_factory.create(port=1111, maxclients=1, admin_port=1112)
+    server.start()
+
+    async with server.client() as client1:
+        assert [b"maxclients", b"1"] == await client1.execute_command("CONFIG GET maxclients")
+
+        with pytest.raises(redis.exceptions.ConnectionError):
+            async with server.client() as client2:
+                await client2.get("test")
+
+        # Check that admin connections are not limited.
+        async with RedisClient(port=server.admin_port) as admin_client:
+            await admin_client.get("test")
+
+        await client1.execute_command("CONFIG SET maxclients 3")
```

Comment:
I was thinking of adding a `get maxclients` here but it's not really needed since you implicitly test the limit on the test case below...